Harfbuzz's CMake config uses the wrong path

### Describe the bug
Trying to use harfbuzz in a CMake project, CMake fails with the following:

```
CMake Error in CMakeLists.txt:
  Imported target "harfbuzz::harfbuzz" includes non-existent path

    "/nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev//nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/include/harfbuzz"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

That path comes from /nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/lib/cmake/harfbuzz/harfbuzz-config.cmake, probably on line 25 which says `set(_harfbuzz_includedir "${prefix}//nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/include")`.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Make a CMake project that tries to link an executable to harfbuzz:
    ```
    find_package(harfbuzz REQUIRED)
    add_executable(program main.c)
    target_link_libraries(program harfbuzz::harfbuzz)
    ```
3. Try to configure project (`cmake .`)
4. See error above

### Expected behavior
The project configures successfully.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@edolstra 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.49, NixOS, 22.11 (Raccoon), 22.11.20220627.020c740`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

--- a/pkgs/development/libraries/harfbuzz/default.nix
+++ b/pkgs/development/libraries/harfbuzz/default.nix
@@ -32,22 +32,12 @@
 
 stdenv.mkDerivation rec {
   pname = "harfbuzz${lib.optionalString withIcu "-icu"}";
-  version = "5.3.1";
+  version = "6.0.0";
 
   src = fetchurl {
     url = "https://github.com/harfbuzz/harfbuzz/releases/download/${version}/harfbuzz-${version}.tar.xz";
-    sha256 = "sha256-Smzgl7dagSH6zEuoO1sIO/7GV/RbADzVo0JPKua0Q00=";
+    sha256 = "HRAQoXUdB21SkeQzwThQKnlNZ5p0mNEmjuIeLUoUDrQ=";
   };
-
-  patches = [
-    # Pick upstream patch for exported symbol test failing on darwin
-    # https://github.com/harfbuzz/harfbuzz/issues/3850
-    (fetchpatch {
-      name = "harfbuzz-fix-check-symbol-tests-lto-and-darwin.patch";
-      url = "https://github.com/harfbuzz/harfbuzz/commit/b0b7a65388da25ae3fa01e969ad6abc67eed4f49.patch";
-      sha256 = "0my064r88pikw6q70hbgf6hwfkw544b9f5ai73qhn2a3c83jqn06";
-    })
-  ];
 
   postPatch = ''
     patchShebangs src/*.py test

